src/components/swap/SwapButtons/SwapButtons.tsx

```diff
@@ -82,6 +115,77 @@ export function SwapButtons({
     return <ButtonConnect />
   }
 
+  if (isGnosisProtocolTradeRequireWrap) {
+    if (gnosisProtocolTradeState === GnosisProtocolTradeState.UNKNOWN)
+      setGnosisProtocolState(GnosisProtocolTradeState.WRAP)
+    const isApprovalRequired = approval !== ApprovalState.APPROVED
+
+    const isSwapDisabled =
+      !isValid ||
+      isApprovalRequired ||
+      (priceImpactSeverity > 3 && !isExpertMode) ||
```

Comment:
what does this 3 means? 
Suggestion: we can extract it to a constant and give it a descriptive name.